### ISSUE TITLE
With longer strings, positions are often wrong and unsorted 

### ISSUE THREAD

**Issue by skovsboll** [hugopl/fzy#2]
Here's a couple of examples:

```
Failures:

  1) Fzy positions are sorted when double letters later in string
     Failure/Error: Fzy.search("bookmarks", %w(clear_bookmarks)).first.positions.should eq([6, 7, 8, 9, 10, 11, 12, 13, 14])

       Expected: [6, 7, 8, 9, 10, 11, 12, 13, 14]
            got: [6, 8, 8, 13, 10, 11, 12, 13, 14]

     # spec/fzy_spec.cr:138

  2) Fzy positions are sorted when double letters beginning of string
     Failure/Error: Fzy.search("aandom", %w(aandom_baandom)).first.positions.should eq([0, 1, 2, 3, 4, 5])

       Expected: [0, 1, 2, 3, 4, 5]
            got: [9, 1, 10, 3, 12, 5]

     # spec/fzy_spec.cr:142

  3) Fzy positions favors start of match
     Failure/Error: Fzy.search("andom", %w(andom_random)).first.positions.should eq([0, 1, 2, 3, 4])

       Expected: [0, 1, 2, 3, 4]
            got: [0, 8, 2, 10, 4]
```

**Comment by hugopl** [comment]
Thanks for reporting!

the positions not being sorted are a real problem... the tests #2 and #3 may be because in some situations there are more than one set of positions that would result in the same score, the dynamic programming algorithm just pick one, not sure if it's the case... I need to better check this, anyway, thanks for the tests!

**Comment by hugopl** [comment]
Thanks, found the issues, was a error while porting the positions code.
Fixed by bc015db9c52a2bd7f12209d1969354a03f1f7714

**Comment by skovsboll** [comment]
Brilliant! It works as expected now. 
Thank you for this awesome lib! 😄
